test/functional/test_requires.py

```diff
@@ -0,0 +1,77 @@
+from collections import namedtuple
+
+import pytest
+
+from taskotron_python_versions.requires import (
+    get_versioned_name,
+    check_requires_naming_scheme,
+)
+
+from .common import gpkg
+
+
+Package = namedtuple('Pcakage', 'name')
+
+
+class QueryStub(object):
+
+    """Stub object for dnf repoquery."""
+
+    def __init__(self, data):
+        """Provide data as {<provide_name>: [<package_name>, ..], ..}, where
+            `provide_name`: (str) name expected to be passed to filter method
+            `package_name`: (str) package name providing it in query result
+        """
+        self.data = data
+        self.provides = None
+
+    def filter(self, provides):
+        self.provides = provides
+        return self
+
+    def run(self):
+        if self.provides not in self.data:
+            raise Exception('Running repoquery on {} is not expected'.format(
+                self.provides))
+        package_names = self.data.pop(self.provides)
+        return [Package(name=name) for name in package_names]
+
+
+@pytest.mark.parametrize(('require', 'repoquery', 'expected'), (
+    ('python-foo', None, None),
+    ('python-foo', QueryStub({'python-foo': []}), None),
+    ('python-foo', QueryStub({'python-foo': ['python-foo']}), None),
+    ('python-foo', QueryStub({'python-foo': ['python2-foo']}), 'python2-foo'),
+))
+def test_get_versioned_name(require, repoquery, expected):
+    assert get_versioned_name(require, repoquery) == expected
+
+
+@pytest.mark.parametrize(('pkgglob', 'repoquery'), (
+    ('python2-geoip2*', QueryStub({})),
+    ('pyserial*', QueryStub({})),
+    ('python-peak-rules*', QueryStub({
+        'python-decoratortools': ['python-decoratortools'],
+        'python-peak-util-addons': ['python-peak-util-addons'],
+        'python-peak-util-assembler': ['python-peak-util-assembler'],
+        'python-peak-util-extremes': ['python-peak-util-extremes'],
+    })),
+))
+def test_package_requires_are_correct(pkgglob, repoquery,):
+    assert not check_requires_naming_scheme(gpkg(pkgglob), repoquery)
+    assert not repoquery.data, (
+        'Repoquery was not called for: {}'.format(repoquery.data))
```

Comment:
This asserts all items from `repoquery.data` were popped out? If so, could you please add a comment about that? Because I don't find it very obvious.